### PR TITLE
Update troubleshooting.md

### PR DESCRIPTION
**Pull request by hnarasaki**
Added information from Internal Wiki on Performance Considerations for IQL Usage.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -101,3 +102,44 @@ Don’t use distinct() as a metric with a large amount of data if you are using
 ### Heap memory size
 The number of rows IQL can return on non-streaming queries depends on the heap size allocated to IQL.
 
+## Performance Considerations for IQL Usage
+
+### Query smaller time ranges
+First test your query on a tiny time range and then ramp up to the required range if performance is sufficient. 
+E.g. `FROM example 1h today ...` is a better way to test than `FROM example 180d today ...`.
+
+If you have created a Superset or Aquarium Dashboard, consider the time range. Do you really need to see 90d today worth of data, or is 2w today sufficient?
```
Comment (youknowjack) [inline docs/troubleshooting.md]: These references are to Indeed-internal tools and need to be removed. I think this whole paragraph can go.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -101,3 +102,44 @@ Don’t use distinct() as a metric with a large amount of data if you are using
...
+
+Try to put the largest grouping as the last grouping. This allows the result of the last largest grouping to be streamed instead of all stored in memory. E.g., since there are dozens of somethings but millions of unique queries, prefer **`GROUP BY something, q`** to **`GROUP BY q, something`** or **~~`GROUP BY q[500000], something[50]`~~**.
+
+### Avoid distinct() when possible
+Avoid distinct() when possible for heavy queries. The optimization of streaming the data in the last group by grouping doesn't work if you have any distinct() elements in SELECT. So `FROM example 1h today select DISTINCT(stringField)` is fine but `FROM example 4w today GROUP BY q SELECT DISTINCT(stringField)` is bad.
+
+Although the output of DISTINCT(field) is a single number, all the values for that field have to be streamed from Imhotep to IQL, which can get really large over larger time ranges for fields with a lot of unique terms (e.g. myStringField, myExample, etc.)
+
+This also applies to IQL2 functions `SUM_OVER()`, `AVG_OVER()` and `DISTINCT_WINDOW()`.
```
Comment (youknowjack) [inline docs/troubleshooting.md]: IQL2 is not available in OSS version yet. Remove this line.

File: docs/troubleshooting.md (hunk 1 above)
Comment (tate-indeed) [reply docs/troubleshooting.md]: @hnarasaki actually already removed this line in pull #34, which was also merged

File: docs/troubleshooting.md (hunk 2 above)
Comment (tate-indeed) [reply docs/troubleshooting.md]: I handled this in #36 .

**Pull request merged**